lib/Twig/Loader/Filesystem.php

```diff
@@ -181,6 +181,12 @@ protected function findTemplate($name, $throw = true)
             return $this->cache[$name];
         }
 
+        if (is_file($name)) {
+            $this->cache[$name] = $name;
+
+            return $name;
```

Comment:
returning a name without the validation is a no-go. It is a security issue. This logic must go after the validation